rust/parquet/src/arrow/array_reader.rs

```diff
@@ -745,16 +1152,66 @@ impl<'a> TypeVisitor<Option<Box<dyn ArrayReader>>, &'a ArrayReaderBuilderContext
     }
 
     /// Build array reader for list type.
-    /// Currently this is not supported.
     fn visit_list_with_item(
         &mut self,
-        _list_type: Rc<Type>,
+        list_type: Rc<Type>,
         _item_type: &Type,
-        _context: &'a ArrayReaderBuilderContext,
+        context: &'a ArrayReaderBuilderContext,
     ) -> Result<Option<Box<dyn ArrayReader>>> {
-        Err(ArrowError(
-            "Reading parquet list array into arrow is not supported yet!".to_string(),
-        ))
+        let mut new_context = context.clone();
+
+        let list_child = &list_type.get_fields()[0];
+        let item_child = &list_child.get_fields()[0];
```

Comment:
This doesn't look correct. I think we should use the `_item_type` passed in for item type. List type in Parquet can be represented with several different kinds of schema as mentioned [here](https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#lists). [This method](https://github.com/apache/arrow/blob/master/rust/parquet/src/schema/visitor.rs#L35) already helped to setup the list item type for you.

Reply:
Yes, that makes more sense. I'm not sure why I did it that way before

Reply:
The failing test made me realize why I had it that way. Currently item_type is passed as `&Type` to `visit_list_with_item` while list_type is passed as `Rc<Type>`. Because item_type is not passed as `Rc<Type>`, if `item_type` is used then when the list items are visited by PrimitiveArrayReader, they will not satisfy the `self.is_included` check, because that checks using the reference. 

I've now changed the signature of `visit_list_with_item` to pass item_type as `Rc<Type>` to fix this issue.